internal/pkg/service/common/etcdop/iterator/iterator.go

```diff
@@ -251,7 +251,7 @@ func nextPageOp(start, end string, pageSize int, revision int64) op.GetManyOp {
 
 	// Ensure atomicity
 	if revision > 0 {
-		opts = append(opts, etcd.WithRev(revision))
+		opts = append(opts, etcd.WithRev(revision), etcd.WithSerializable())
```

Comment:
Optimalizacia strankovania

-----

Etcd ma 2 grarancie konzistencie:
 - https://etcd.io/docs/v3.5/learning/api_guarantees/#isolation-level-and-consistency-of-replicas
 - linearizable read, default, quorum read
 - serializable read, faster, single node read

----------
 
- `linearizable` zarucuje ze `n > nodes/2` (quorum) nodes ma danu verziu ulozenu na disku, pri citani sa tak musi kontaktovat cely cluster, ak odpovie nadpolovicna vacsina, tak sa hodnota vrati.
- `serializable` sa tyka iba jednej node, ak je tam dana verzia pritomna, tak sa hodnota vrati

-----------



- Iterator cita prvu page bez `revision` a `WithSerializable` - tj. pouzije sa default `linearizable` read.
  - Quorum sa zhodne, ze `123` je najnovsia vezia.
- Dalsie stranky chceme z rovnakeho okamihu v case, z rovnakej `revision`, pri prvej stranke sme mali garantovane, ze mame aktualnu verziu, tak pri dalsich stankach uz nemusime kontaktovat cely cluster, pouzijeme `WithSerializable`.
 
 ----
 
 Podobne najdeme tuto kombinaciu aj na viacerych miestach v `etcd client`:
 https://github.com/etcd-io/etcd/blob/8b93e0e46bb0a698e425d8a4b0ba82f7510fd0c8/client/v3/concurrency/stm.go#L320-L321